temporal-sdk/src/main/java/io/temporal/internal/common/WorkflowExecutionUtils.java

```diff
@@ -229,15 +234,17 @@ public static HistoryEvent getInstanceCloseEvent(
               .setWaitNewEvent(true)
               .setNextPageToken(pageToken)
               .build();
+
       long elapsed = System.currentTimeMillis() - start;
-      Deadline expiration = Deadline.after(unit.toMillis(timeout) - elapsed, TimeUnit.MILLISECONDS);
-      if (expiration.timeRemaining(TimeUnit.MILLISECONDS) > 0) {
+      long millisRemaining = unit.toMillis(timeout != 0 ? timeout : Integer.MAX_VALUE) - elapsed;
```

Comment:
`timeout : Integer.MAX_VALUE` this default wasn't there before, why do we need it?

Reply:
It's kinda a mess right now. This line didn't accept 0 as timeout, but two lines after it we actually have a check timeout != 0:
`if (timeout != 0 && System.currentTimeMillis() - start > unit.toMillis(timeout))`

So, I added javadoc explicitly stating that 0 is supported and added this processing. It's not needed because right now we provide timeout this way usually:
https://github.com/temporalio/sdk-java/blob/fd28e214fe9820e00a3f94883b3faaa2e27ae63e/temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowStubImpl.java#L261

It's also done inconsistently. Some places provide MAX_INT, some MAX_LONG

Reply:
Yeah, there is even that https://github.com/temporalio/sdk-java/blob/fd28e214fe9820e00a3f94883b3faaa2e27ae63e/temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowStubImpl.java#L225 where max int is used intentionally.